### PR TITLE
Add missing error check in posix TCP handling

### PR DESCRIPTION
**Pull request by Kswin01**
This PR properly enables TCP keep-alive for lwip. Additionally we add proper error checking to the posix layer, where before we were not checking the return value of tcp socket create and assigning bad socket handles.

### CODE REVIEW

File: components/fs/nfs/tcp.c [hunk 1]
```diff
@@ -357,11 +359,12 @@ err_t socket_connected(void *arg, struct tcp_pcb *tpcb, err_t err)
     assert(socket->state == socket_state_connecting);
 
     socket->state = socket_state_connected;
-    
+
     tcp_sent(tpcb, socket_sent_callback);
     tcp_recv(tpcb, socket_recv_callback);
 
     tpcb->so_options |= SOF_KEEPALIVE;
+    socket->sock_tpcb->keep_intvl = 1000;
```
Comment (Ivan-Velickovic) [inline components/fs/nfs/tcp.c:367]: Can we have something like `#define LWIP_KEEPALIVE_INTERVAL 1000` in `lwipopts.sh`?

Comment (Ivan-Velickovic) [reply components/fs/nfs/tcp.c:367]: And also a comment next to the define that says it's 10s etc.

File: components/fs/nfs/lwip_include/lwipopts.h [hunk 2]
```diff
@@ -21,7 +21,9 @@
 #define LWIP_ICMP 1
 #define LWIP_RAND rand
 #define LWIP_DHCP 1
-
+#define LWIP_TCP_KEEPALIVE  1
+#define TCP_KEEPIDLE_DEFAULT    1000
```
Comment (Ivan-Velickovic) [inline components/fs/nfs/lwip_include/lwipopts.h:25]: Should this be called `TCP_KEEPINTVL_DEFAULT` instead?

**Comment by wom-bat** [comment]
> I'm not sure we want to change these so dramatically without proper justification. It might incidentally fix an issue but not fix the real cause. Do you know anything about appropriate times for keepalive @wom-bat?

Standard says between 10 minutes and 2 hours.  I'd suggest around 15 minutes for our use cases.
See https://webhostinggeeks.com/howto/tcp-keepalive-recommended-settings-and-best-practices/

**Comment by Ivan-Velickovic** [comment]
After talking to @JE-Archer we'll just merge the patch that adds the error check. @JE-Archer believes there is a deeper issue and playing with the parameters of lwIP is just hiding that, so he's looking into that.

**Review by Ivan-Velickovic — APPROVED** [review]

**Pull request merged**